trade-finance-manager-ui/server/middleware/validatePostAddPaymentRequestBody/index.ts

```diff
@@ -20,6 +35,10 @@ const redirectWithError = (
 ) => {
   req.session.addPaymentErrorKey = addPaymentError;
   req.session.checkedCheckboxIds = checkedCheckboxIds;
+  const facilityIdQuery = getFacilityIdQueryFromReferer(req);
+  if (facilityIdQuery) {
+    return res.redirect(`/utilisation-reports/${reportId}?facilityIdQuery=${facilityIdQuery}`);
+  }
```

Comment:
To avoid this `if` block, could we use `axios.getUri` to construct this url? I just checked and, if `facilityIdQuery` is `undefined`, `getUri` will not include it in the url which matches the behaviour we want